lib/logstash/runner.rb

```diff
@@ -27,6 +28,9 @@ def main(args)
       return 1
     end
 
+    # Print a warning to STDERR for bad java versions
+    LogStash::Util::JavaVersion.warn_on_bad_java_version
```

Comment:
why not change this into
`LogStash::Util::JavaVersion.warn_bad_java_version if bad_java_version?`
Then this predicate can return false on MRI